src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -504,6 +539,32 @@ private static MethodInfo SelectTargetMethod(Type targetType, MethodInfo proxyMe
                     continue;
                 }
 
+                // Check if the candidate method is a reverse mapped method
+                DuckReverseMethodAttribute reverseMethodAttribute = candidateMethod.GetCustomAttribute<DuckReverseMethodAttribute>(true);
+                if (reverseMethodAttribute?.Arguments is not null)
```

Comment:
`reverseMethodAttribute?.Arguments is not null` will also evaluate to `null` for a reverse proxy method that takes no arguments. Is that intentional? Would we ever have a need to define a reverse proxy method like that?

Reply:
Yeah, the `Arguments` attribute intended use is for 2 cases:
1) Multiple overloads with same number of attributes with arguments types that we don't have at compile time (ex: custom type).
2) To enable ducktyping over the arguments.

So is optional, if you don't define the arguments the normal `SelectTargetMethod` algorithm will try to select the best match for the one you are trying to ducktype.

Reply:
For the second case here you have an example: https://github.com/DataDog/dd-trace-dotnet/pull/1402/files#diff-fb46ebabc7e0a000432c1d76a6dc06a298ffeef5795e6ff2edbae76c25c11ff8R54 . ILogEvent will be the ducktype interface for `Datadog.Trace.Vendors.Serilog.Events.LogEvent`